File: Selenium/Activities/Activity19.java
/*
 Open a new browser to https://training-support.net/webelements/alerts
Get the title of the page and print it to the console.
Find the button to open a CONFIRM alert and click it.
Switch the focus from the main window to the Alert box and get the text in it and print it.
Close the alert once with Ok and again with Cancel.
Close the browser.
 */

package training;

import org.openqa.selenium.Alert;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;

public class Activity19 {

	public static void main(String[] args) {
		
		//Create an instance of the firefox driver
		WebDriver driver = new FirefoxDriver();
		
		//Open the browser and print the title of the web page
		driver.get("https://training-support.net/webelements/alerts");
		System.out.println("Title of the web page: " + driver.getTitle());
		
		//Find the button to open a CONFIRM alert and click it
		driver.findElement(By.id("confirmation")).click();
		
		//Switch the focus from the main window to the alert box
		Alert confirmAlert = driver.switchTo().alert();
		
		//Get the text in the alert and print it
		System.out.println("Text in the alert: " + confirmAlert.getText());
		
		//Close the alert by clicking ok
		confirmAlert.accept();
		
		//Can also be closed by clicking cancel
		//confirmAlert.dismiss();
		
		//Print the message
		System.out.println(driver.findElement(By.id("result")).getText());
		
		//Close the browser
		driver.quit();
	}
}

File: Selenium/Activities/Activity19.py
""" Open a new browser to https://training-support.net/webelements/alerts
Get the title of the page and print it to the console.
Find the button to open a CONFIRM alert and click it.
Switch the focus from the main window to the Alert box and get the text in it and print it.
Close the alert once with Ok and again with Cancel.
Close the browser."""

#Import the web driver
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

#Initialize the firefox driver
with webdriver.Firefox() as driver:
    #Initialize the wait variable
    wait = WebDriverWait(driver, timeout=10)

    #Open the browser and print the title of the page
    driver.get("https://training-support.net/webelements/alerts")
    print("Title of the web page: ", driver.title)

    #Find the alert confirmation and click it
    driver.find_element(By.ID, "confirmation").click()

    #Switch to the alert box
    confirmAlert = wait.until(EC.alert_is_present())

    #Print the text in the alert
    alertText = confirmAlert.text
    print("Text in the alert: " + alertText)

    #Close the alert by clicking ok
    confirmAlert.dismiss()

    #Print the message
    print(driver.find_element(By.ID, "result").text)